src/Support/Payload.php

```diff
@@ -19,33 +19,158 @@ class Payload {
 		$term = get_term( $term_id );
-		return array( 'taxonomy' => $term->taxonomy );
+		
+		// Handle case where term doesn't exist or is an error
+		if ( is_wp_error( $term ) || ! $term ) {
+			return array( 'taxonomy' => '' );
+		}
+
+		$taxonomy = $term->taxonomy;
+		$payload  = array( 'taxonomy' => $taxonomy );
+
+		// Add REST API URL if taxonomy has REST support enabled
+		if ( self::taxonomy_supports_rest( $taxonomy ) ) {
+			$payload['rest_url'] = self::get_term_rest_url( $term_id, $taxonomy );
+		}
+
+		return $payload;
 	}
 
 	/**
 	 * Generate payload data for a user.
 	 *
 	 * @param int $user_id The user ID.
-	 * @return array<string,mixed> The payload data containing user roles.
+	 * @return array<string,mixed> The payload data containing user roles and REST URL if supported.
 	 */
 	public static function user( int $user_id ): array {
-		$user  = get_userdata( $user_id );
-		$roles = ( $user && $user->roles ) ? array_values( $user->roles ) : array();
+		$user    = get_userdata( $user_id );
+		$roles   = ( $user && $user->roles ) ? array_values( $user->roles ) : array();
+		$payload = array( 'roles' => $roles );
+
+		// Add REST API URL if users endpoint has REST support enabled
+		if ( self::users_support_rest() ) {
+			$payload['rest_url'] = self::get_user_rest_url( $user_id );
+		}
+
+		return $payload;
+	}
+
+	/**
+	 * Check if a post type supports REST API.
+	 *
+	 * @param string|false $post_type The post type to check.
+	 * @return bool Whether the post type supports REST API.
+	 */
+	private static function post_type_supports_rest( $post_type ): bool {
+		if ( false === $post_type ) {
+			return false;
+		}
+
+		$post_type_object = get_post_type_object( $post_type );
+		return $post_type_object && $post_type_object->show_in_rest;
+	}
 
-		return array( 'roles' => $roles );
+	/**
+	 * Check if a taxonomy supports REST API.
+	 *
+	 * @param string $taxonomy The taxonomy to check.
+	 * @return bool Whether the taxonomy supports REST API.
+	 */
+	private static function taxonomy_supports_rest( string $taxonomy ): bool {
+		$taxonomy_object = get_taxonomy( $taxonomy );
+		return $taxonomy_object && $taxonomy_object->show_in_rest;
+	}
+
+	/**
+	 * Check if users endpoint supports REST API.
+	 *
+	 * @return bool Whether users support REST API.
+	 */
+	private static function users_support_rest(): bool {
+		// Users REST API is enabled by default in WordPress but can be disabled
+		// Check if the WP_REST_Users_Controller is available
+		return class_exists( 'WP_REST_Users_Controller' );
+	}
+
+	/**
+	 * Generate REST API URL for a post.
+	 *
+	 * @param int          $post_id   The post ID.
+	 * @param string|false $post_type The post type.
+	 * @return string The REST API URL for the post.
+	 */
+	private static function get_post_rest_url( int $post_id, $post_type ): string {
+		if ( false === $post_type ) {
+			return '';
+		}
+
+		$post_type_object = get_post_type_object( $post_type );
```

Comment:
Rework to build url as well as rest check from these post_type properties:

/**
	 * Whether this post type should appear in the REST API.
	 *
	 * Default false. If true, standard endpoints will be registered with
	 * respect to $rest_base and $rest_controller_class.
	 *
	 * @since 4.7.4
	 * @var bool $show_in_rest
	 */
	public $show_in_rest;

	/**
	 * The base path for this post type's REST API endpoints.
	 *
	 * @since 4.7.4
	 * @var string|bool $rest_base
	 */
	public $rest_base;

	/**
	 * The namespace for this post type's REST API endpoints.
	 *
	 * @since 5.9.0
	 * @var string|bool $rest_namespace
	 */
	public $rest_namespace;